### ISSUE TITLE
Problem with Shell back button

### ISSUE THREAD

**Issue by Senso4sLab** [dotnet/maui#9604]
### Description

I added in the Shell template project the following code to the Content page
```
<Shell.BackButtonBehavior >
        <BackButtonBehavior  IconOverride="back_icon.png"/>
 </Shell.BackButtonBehavior>
```
If I press the back button the following expectation is thrown:
Java.Lang.IllegalArgumentException: 'No drawer view found with gravity LEFT' 

### Steps to Reproduce

1. Create a new Shell template project
2.  Add BackButtonbehavior in the Content page.
3.  Press the back button and a surprise will appear

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

**Comment by Senso4sLab** [comment]
Here is the repo:

https://github.com/Senso4sLab/ShellBackButtonProblem.git

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 4.0. Can repro on android platform with sample project.
[ShellBackButton.zip](https://github.com/dotnet/maui/files/12178765/ShellBackButton.zip)
![Screenshot 2023-07-27 102928](https://github.com/dotnet/maui/assets/77434865/23e9ca3c-444e-45fc-960a-48857589a65d)

**Comment by jfversluis** [comment]
I think this might be fixed with #20087
